src/components/Creations/Creations.tsx

```diff
@@ -40,59 +42,69 @@ const Creations = (props: Props) => {
         : Categories.WEARABLES,
     [filters.category]
   )
+  const selectedRarities = useMemo(() => filters.rarities?.split(',') ?? [], [filters.rarities])
 
   useEffect(() => {
     const shouldLoadMultiplePages = !count && page !== 1
     onFetchCreations({
       creator: props.profileAddress,
       first: shouldLoadMultiplePages ? page * ITEMS_PER_PAGE : first,
       skip: shouldLoadMultiplePages ? 0 : ITEMS_PER_PAGE * (page - 1),
-      category: selectedCategory
+      category: selectedCategory,
+      rarities: selectedRarities as Rarity[]
     })
   }, [page, first, filters, sortBy])
 
   const categories = useMemo(() => buildCategoryFilterCategories(), [])
-  const onChangeCategory = useCallback((value: string) => {
-    changeFilter('category', value)
-  }, [])
+  const onChangeCategory = useCallback(
+    (value: string) => {
+      changeFilter('category', value)
+    },
+    [changeFilter]
+  )
+  const onChangeFilter = useCallback(
+    (value: string[]) => {
+      changeFilter('rarities', value.join(','))
```

Comment:
why does this need to be only a string? can't we accept string[] in the changeFilter function

Reply:
If we accept arrays in the `changeFilter` method, we'll also need to accept query string arrays when reading the filters as well, which would make every parameter of type `string | string[] | null`, requiring us to do an extra validation when using any of the filters. As this is the only filter at the moment that has the selection of multiple parameters, it was easier to accept a formatted list of rarities.
Do you think that it would be better to change it?

Reply:
Its ok we can leave like this for now :)